### ISSUE TITLE
Fatal Error - Ubuntu 20.04 - Terraform 1.1.7

### ISSUE THREAD

**Issue by zkhan4176** [hashicorp/terraform#30686]
Hi,

I am trying to install terraform on ubuntu 20.04 (focal). I followed the following link

https://learn.hashicorp.com/tutorials/terraform/install-cli

and followed "linux->ubunutu/debian" tab. At the end of installation when I run following command

```terraform -help```

I get the error

```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x80 addr=0x0 pc=0x41890c]

goroutine 1 [running, locked to thread]:
runtime.throw(0x289f968, 0x2a)
	runtime/panic.go:1117 +0x72 fp=0xc0001d9400 sp=0xc0001d93d0 pc=0x438412
runtime.sigpanic()
	runtime/signal_unix.go:718 +0x2e5 fp=0xc0001d9438 sp=0xc0001d9400 pc=0x44f105
runtime.heapBitsSetType(0xc0001b5c70, 0xd0, 0xd0, 0x25e90a0)
	runtime/mbitmap.go:1112 +0x26c fp=0xc0001d9518 sp=0xc0001d9438 pc=0x41890c
runtime.mallocgc(0xd0, 0x25e90a0, 0x315ecd70d06c9101, 0x3d78ba0)
	runtime/malloc.go:1096 +0x5c5 fp=0xc0001d95a0 sp=0xc0001d9518 pc=0x40fa25
runtime.newobject(...)
	runtime/malloc.go:1177
runtime.mapassign_faststr(0x2460c60, 0xc00024d320, 0x2834881, 0x4, 0xc0000ebf2d)
	runtime/map_faststr.go:220 +0x38b fp=0xc0001d9608 sp=0xc0001d95a0 pc=0x41642b
github.com/hashicorp/terraform/internal/terraform.init()
	github.com/hashicorp/terraform/internal/terraform/node_resource_validate.go:159 +0x235 fp=0xc0001d9658 sp=0xc0001d9608 pc=0xc37895
runtime.doInit(0x3c273a0)
	runtime/proc.go:6309 +0xec fp=0xc0001d97a8 sp=0xc0001d9658 pc=0x44886c
runtime.doInit(0x3c1ec60)
	runtime/proc.go:6286 +0x72 fp=0xc0001d98f8 sp=0xc0001d97a8 pc=0x4487f2
runtime.doInit(0x3c24540)
	runtime/proc.go:6286 +0x72 fp=0xc0001d9a48 sp=0xc0001d98f8 pc=0x4487f2
runtime.doInit(0x3c1d260)
	runtime/proc.go:6286 +0x72 fp=0xc0001d9b98 sp=0xc0001d9a48 pc=0x4487f2
runtime.doInit(0x3c25080)
	runtime/proc.go:6286 +0x72 fp=0xc0001d9ce8 sp=0xc0001d9b98 pc=0x4487f2
runtime.doInit(0x3c35dc0)
	runtime/proc.go:6286 +0x72 fp=0xc0001d9e38 sp=0xc0001d9ce8 pc=0x4487f2
runtime.doInit(0x3c28080)
	runtime/proc.go:6286 +0x72 fp=0xc0001d9f88 sp=0xc0001d9e38 pc=0x4487f2
runtime.main()
	runtime/proc.go:208 +0x205 fp=0xc0001d9fe0 sp=0xc0001d9f88 pc=0x43ac05
runtime.goexit()
	runtime/asm_amd64.s:1371 +0x1 fp=0xc0001d9fe8 sp=0xc0001d9fe0 pc=0x46f8a1
```

**Comment by apparentlymart** [comment]
Hi @zkhan4176! Sorry for this crash, and thanks for reporting it.

Unfortunately this stack trace is quite confusing since it seems to be happening during the Go runtime's own initialization, before the main Terraform application code is even running. It does indicate that it's running some application-defined initialization code in the main `terraform` package, but curiously the indicated line number in `node_resource_validate.go` refers to a comment in v1.1.7, not to an `init` function as the stack trace suggests:

https://github.com/hashicorp/terraform/blob/2a5420cb9acf8d5f058ad077dade80214486f1c4/internal/terraform/node_resource_validate.go#L159

Given what appears in the rest of the stack trace, I suspect that this is a hidden `init` function generated automatically by the Go compiler in order to create the value to assign to a global variable, rather than an explicit `init` function that we wrote, but I'm not sure how to understand _which_ variable the Go runtime was trying to initialize at the point where it encountered this error.

Unfortunately in order to make progress here I think we'll need to be able to reproduce this locally so that we can attach a debugger to Terraform to try to see exactly what the Go runtime was trying to initialize when this occurred, and why the memory allocator encountered a segmentation violation while trying to initialize it. This seems like it will be hard to reproduce because it probably relates to something specific to how your system is configured, which we would not then be able to reproduce. :confounded: 

Probably the best we could do with the information already here is identify the upstream Go commit that we built Terraform CLI v1.1.7 from and examine the relevant `runtime/malloc.go` and `runtime/mbitmap.go` lines mentioned in the stack trace to see if that gives a clue as to what the Go runtime was doing at the time it encountered this problem.

Thanks again for reporting this! I'm afraid I don't have any specific suggestions for how to work around this problem at the moment, because it is not a behavior we have seen before.
